pkg/graph/resolvers/system_intake_status_requester.go

```diff
@@ -0,0 +1,107 @@
+package resolvers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// CalculateSystemIntakeRequesterStatus calculates the status to display in the requester view for a System Intake request,
+// based on the intake's current step, the state of that step, and the overall intake state (open/closed)
+func CalculateSystemIntakeRequesterStatus(intake *models.SystemIntake, currentTime time.Time) (models.SystemIntakeStatusRequester, error) {
+	if intake.State == models.SystemIntakeStateCLOSED && intake.DecisionState == models.SIDSNoDecision {
+		return models.SISRClosed, nil
+	}
+
+	switch intake.Step {
+	case models.SystemIntakeStepINITIALFORM:
+		return calcSystemIntakeInitialFormStatusRequester(intake.RequestFormState)
+	case models.SystemIntakeStepDRAFTBIZCASE:
+		return calcSystemIntakeDraftBusinessCaseStatusRequester(intake.DraftBusinessCaseState)
+	case models.SystemIntakeStepGRTMEETING:
+		// this calc function doesn't use a switch statement and can't possibly return an error
+		return calcSystemIntakeGRTMeetingStatusRequester(intake.GRTDate, currentTime), nil
+	case models.SystemIntakeStepFINALBIZCASE:
+		return calcSystemIntakeFinalBusinessCaseStatusRequester(intake.FinalBusinessCaseState)
+	case models.SystemIntakeStepGRBMEETING:
+		// this calc function doesn't use a switch statement and can't possibly return an error
+		return calcSystemIntakeGRBMeetingStatusRequester(intake.GRBDate, currentTime), nil
+	case models.SystemIntakeStepDECISION:
+		return calcSystemIntakeDecisionStatusRequester(intake.DecisionState)
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid step")
```

Comment:
Defaulting to an error is probably the right way to handle all these switch statements, nice job.